src/fs/metadata_log_operations/write.hh

```diff
@@ -223,6 +224,8 @@ private:
         assert(aligned_expected_write_len % _metadata_log._alignment == 0);
         assert(disk_buffer->bytes_left() >= aligned_expected_write_len);
 
+        _metadata_log.throw_if_read_only_fs();
+
```

Comment:
Again, throwing exception instead of returning it in async function (however, IDK if it is a bug).

Reply:
If I understand it correctly it should work as intended because it is called in continuation of repeat() from line 161 and that exception would be converted into exception_future. I could change it to throw_exception_future_if_read_only_fs() but it would add another tab and it is very unlikely that it would throw anyway so I would leave it as it is. What do you think?

Reply:
@psarna What do you think?

Reply:
Throwing is technically inside continuations, since they will get caught and translated to exception futures, **but**, it's less optimal, since the C++ library implementation of exceptions has overhead, so it's more optimal to return an exception future.

Now, one thing that I cannot accept is naming this function **throw**_exception_future_if_read_only_fs - it doesn't throw anything, it simply returns an exceptional future (which is a correct thing to do). The 'throw' part in its name may suggest to people that they need to wrap it in try-catch. `return_exception_if_read_only_fs()` would be fine (future is kind of implicit in seastar, so it can be dropped from the name to make it shorter).

Reply:
That said, creating a new continuation just to represent something like that:
```cpp
return return_exception_if_read_only_fs().then([] {
    // function body
});
```
also has a cost - additional continuations mean additional allocations, and we don't want unnecessary allocations on the happy path.

So, to sum up: please change the name of the function that **returns** an exception future, but also, in this particular case, just call the one that throws - throwing is considered to be rare, so it's better to save a continuation than to always create one in case the system became read only.

Reply:
In that case do we need return_exception_if_read_only_fs() at all? For now, it is used only in schedule_flush_of_curr_cluster() but it could be removed too.

Reply:
If this is the only place and the reasons above make it almost useless, I would remove it.